src/EllCrv/QQ.jl

```diff
@@ -871,6 +1155,84 @@ function tates_algorithm_global(E)
   return E::EllCrv{fmpq}
 end
 
+function tamagawa_number(E::EllCrv{fmpq},p)
+  return tates_algorithm_local(E,p)[4]
+end
+
+function tamagawa_numbers(E::EllCrv{fmpq})
+  badp = bad_primes(E)
+  return [tamagawa_number(E,p) for p in badp]
+end
+
+function kodaira_symbol(E::EllCrv{fmpq},p)
+  return tates_algorithm_local(E,p)[2]
+end
+
+function kodaira_symbols(E::EllCrv{fmpq})
+  badp = bad_primes(E)
+  return [kodaira_symbol(E,p) for p in badp]
+end
+
+
+function tamagawa_number(E::EllCrv{nf_elem},p::NfOrdIdl)
+  return tates_algorithm_local(E,p)[4]
+end
+
+function tamagawa_numbers(E::EllCrv{nf_elem})
+  badp = bad_primes(E)
+  return [tamagawa_number(E,p) for p in badp]
+end
+
+function kodaira_symbol(E::EllCrv{nf_elem},p::NfOrdIdl)
+  return tates_algorithm_local(E,p)[2]
+end
+
+function kodaira_symbols(E::EllCrv{nf_elem})
+  badp = bad_primes(E)
+  return [kodaira_symbol(E,p) for p in badp]
+end
+
+function reduction_type(E::EllCrv{fmpq}, p)
+  Ep, Kp, f, c, split = tates_algorithm_local(E, p)
+  
+  if Kp=="I0"
+    return "Good"
+  end
+  
+  if match(r"(I)([0-9]*)", Kp).match == Kp
+    if split
+      return "Split Multiplicative"
+    else
+      return "Nonsplit multiplicative"
+    end
+  end
+ 
+ return "Additive"
+  
+end
+  
+
+function reduction_type(E::EllCrv{nf_elem}, p::NfOrdIdl)
+  Ep, Kp, f, c, split = tates_algorithm_local(E, p)
+  
+  if Kp=="I0"
+    return "Good"
+  end
+  
+  if match(r"(I)([0-9]*)", Kp).match == Kp
+    if split
+      return "Split Multiplicative"
+    else
+      return "Nonsplit multiplicative"
+    end
+  end
```

Comment:
Not sure if string matching and using "naked" strings to represent features is a good idea on the long run? Perhaps OK for a prototype, though